sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -207,12 +207,12 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
       messageParameters = Map("typeName" -> (dataType + failure)))
   }
 
-  def failedExecuteUserDefinedFunctionError(funcCls: String, inputTypes: String,
+  def failedExecuteUserDefinedFunctionError(functionName: String, inputTypes: String,
       outputType: String, e: Throwable): Throwable = {
     new SparkException(
       errorClass = "FAILED_EXECUTE_UDF",
       messageParameters = Map(
-        "functionName" -> funcCls,
+        "functionName" -> functionName,
```

Comment:
Please, quote the function name as you are here:
```suggestion
        "functionName" -> toSQLId(functionName),
```

Reply:
I added it to be consistent with other usages, but it does look a little weird to me in this context, particularly for the Hive UDF changes where it adds backticks in the class name.

`QueryErrorsBase.toSQLId` passes the input to `UnresolvedAttribute.parseAttributeName`. The `functionName` here is not an attribute name so it doesn't seem like `toSQLId` should not be used here. The main purpose of `toSQLId` that I see is to remove a prefix of `__auto_generated_subquery_name` in an attribute name and backticks between period delimiters. Since this input is a UDF name and its class location it doesn't feel applicable to me.

Reply:
> The main purpose of toSQLId that I see is to remove a prefix of __auto_generated_subquery_name in an attribute name and backticks between period delimiters.

Main purpose of quoting is to highlight inlined (variable) elements in the error message format. In particular, we mark SQL identifiers in error messages by backticks.

Is `functionName` an identifier or not in this context?

Reply:
Yes `functionName` is acting as an identifier for a UDF here. My main concern is for the hive UDF case the backticks around each part of the fully qualified class name does not feel natural. I would prefer the backticks around the entire name like `` `org.apache.foo.Bar` ``. This is subjective though so I'm okay deferring to your preference.